### ISSUE TITLE
Investigate a possible issue with find_account_ids for AWS ALB integration

### ISSUE THREAD

**Issue by CarlosRS9** [wazuh/wazuh#14079]
| Affected integration |
|---|
| AWS ALB|

## Description

A user has reported us through our slack community a possible bug with the AWS ALB integration. He indicates that he has several ALB logs, stored in a bucket, which are correctly processed using Wazuh 4.2, but when trying to process them using Wazuh 4.3 the module fails, throwing the following error:

```
2022/06/29 09:54:54 wazuh-modulesd:aws-s3[25445] wm_aws.c:126 at wm_aws_main(): INFO: Executing Bucket Analysis: (Bucket: continuum-uat-alb-private-log, Type: alb)
2022/06/29 09:54:54 wazuh-modulesd:aws-s3[25445] wm_aws.c:318 at wm_aws_run_s3(): DEBUG: Create argument list
2022/06/29 09:54:54 wazuh-modulesd:aws-s3[25445] wm_aws.c:433 at wm_aws_run_s3(): DEBUG: Launching S3 Command: wodles/aws/aws-s3 --bucket alblog1 --access_key ACCESSKEY --secret_key SECRETKEY --only_logs_after 2022-Jun-26 --type alb --debug 2 --skip_on_error
2022/06/29 09:54:55 wazuh-modulesd:aws-s3[25445] wm_aws.c:447 at wm_aws_run_s3(): WARNING: Bucket:  -  Returned exit code 12
2022/06/29 09:54:55 wazuh-modulesd:aws-s3[25445] wm_aws.c:466 at wm_aws_run_s3(): WARNING: Bucket:  -  Invalid type of bucket. The bucket was set up as 'alb' type and this bucket does not contain log files from this type. Try with other type: {'vpcflow', 'custom', 'cloudtrail', 'config', 'guardduty', 'waf'}

2022/06/29 09:54:55 wazuh-modulesd:aws-s3[25445] wm_aws.c:471 at wm_aws_run_s3(): DEBUG: Bucket:  -  OUTPUT: DEBUG: +++ Debug mode on - Level: 2
ERROR: Invalid type of bucket. The bucket was set up as 'alb' type and this bucket does not contain log files from this type. Try with other type: {'vpcflow', 'custom', 'cloudtrail', 'config', 'guardduty', 'waf'}

2022/06/29 09:54:55 wazuh-modulesd:aws-s3[25445] wm_aws.c:166 at wm_aws_main(): INFO: Fetching logs finished.
```

No additional useful information is shown when running the module with debug mode enabled. This is the output when running it manually:
```
DEBUG: +++ Debug mode on - Level: 2
ERROR: Invalid type of bucket. The bucket was set up as 'alb' type and this bucket does not contain log files from this type. Try with other type: {'guardduty', 'custom', 'waf', 'vpcflow', 'cloudtrail', 'config'}
```

During our manual testing using the ALB logs at our disposal the module worked correctly. We need to generate new ALB logs and test processing them to find a possible cause for this discrepancy between 4.2 and 4.3.

## Tasks
- [x] Check if we should generate new ALB logs to be able to replicate the issue.
- [x] Find the root cause of this behaviour and check if this is caused by a bug.
- [x] Determine if this affects the rest of the bucket integrations.

**Comment by CarlosRS9** [comment]
After some investigation we have been able to confirm that there is no bug. The cause of this error was that the user had his logs stored in a path that contained a prefix like the following, but was not indicating this prefix in his configuration:
```
<bucket_name>/prefix/AWSLogs/<account_id>/elasticloadbalancing/<region>/<year>/<month>/<day>
```

The solution is to correctly apply the prefix in the configuration:
```
    <bucket type="alb">
      <name>alblog</name>
      <access_key>ACCESSKEY</access_key>
      <secret_key>SECRETKEY</secret_key>
      <path>your custom prefix</path>
      <only_logs_after>2022-Jun-26</only_logs_after>
    </bucket>
```


The 4.3 integration was failing because it was looking for the `bucket_name/AWSLog/` path and it does not exists. Wazuh 4.2 does not fails in this case because the ALB integration was processing every path available in your bucket, which is not recommended as it could result in processing unwanted logs. This was fixed in 4.3.

We can conclude this is not a bug. It is correct that the module fails in this case if you do not specify the proper prefix. What we need to do is to improve the handling of this error and the message displayed, as it was not clearly indicating the true nature of the error.

On the other hand, the logs we have are completely valid, so it is not necessary to generate new ones for future tests.
